ee/agent/knapsack/knapsack.go

```diff
@@ -239,3 +243,17 @@ func (k *knapsack) CurrentEnrollmentStatus() (types.EnrollmentStatus, error) {
 
 	return types.Enrolled, nil
 }
+
+func (k *knapsack) SetEnrollmentDetails(details types.EnrollmentDetails) error {
+	// Only update if there are actual changes
+	if details != enrollmentDetails {
+		k.slogger.Logger.DebugContext(context.Background(), "updating enrollment details")
```

Comment:
We prefer to use the `Log` method with the corresponding level for consistency. Probably useful to also log the old/new details to see what changed?

```
k.Slogger().Log(context.Background(), slog.LevelDebug,
	"updating enrollment details",
	"old_details", enrollmentDetails,
	"new_details", details,
)
```

(May need to do `fmt.Sprintf("%+v", enrollmentDetails)`, I forget how well these get stringified inside the logs.)